src/Orchard/Environment/Extensions/Folders/IExtensionFolders.cs

```diff
@@ -4,5 +4,6 @@
 namespace Orchard.Environment.Extensions.Folders {
     public interface IExtensionFolders {
         IEnumerable<ExtensionDescriptor> AvailableExtensions();
+        IEnumerable<string> Paths { get; }

```

Comment:
I don't think it's good to have this property on the interface. IExtensionFolders is quite a high-level concept (though naming it "Folders" gives away some clues...) and I wouldn't surface the internally used paths like this (it doesn't really make sense for CoreModuleFolders e.g.). There can be simply a public property like this just on the module and theme folders.